finagle-stats-core/src/main/scala/com/twitter/finagle/stats/NonReentrantReadWriteLock.scala

```diff
@@ -25,8 +26,24 @@ private[stats] final class NonReentrantReadWriteLock extends AbstractQueuedSynch
   // and our use case doesn't require knowing when the lock is held exclusively.
 
   override def tryAcquireShared(num: Int): Int = {
-    val cur = getState()
-    if (cur >= 0 && compareAndSetState(cur, cur + 1)) 1 else -1
+    val SpinMax = 20
+    var ret: Int = -1
+    var cur: Int = 0
+    var tries: Int = 0
+
+    // Try to acquire the lock up to SpinMax times before failing and 
+    // allowing AbstractQueuedSynchronizer to execute the slow-path lock.
+    do {
+      cur = getState()
+      ret = if (cur >= 0 && compareAndSetState(cur, cur + 1)) {
+        1
+      } else {
+        Thread.onSpinWait()
```

Comment:
hmm, looking at this more deeply, it seems like Thread.onSpinWait only shows up in JDK9, but we haven't deprecated JDK8 yet.  How important is this to the ARM implementation?

Reply:
Please let me know if what I did is not going to be accepted on style reasons, but I simply commented it out with an explanation of what to do when JDK8 is deprecated.  `onSpinWait()` can be omitted, it has a small negative effect though because the system is not getting throttled when it should be, so instead `compareAndSetState` gets spammed in a tight loop, making it pause a little in between is good for both x86 and Arm.

Reply:
This looks fine to me, since you left an explanatory comment.